planemo/galaxy/activity.py

```diff
@@ -236,6 +242,47 @@ def _execute(  # noqa C901
         ctx.vlog("collecting outputs from run...")
         run_response.collect_outputs(output_directory)
         ctx.vlog("collecting outputs complete")
+
+    if kwds.get("upload_instance_url") or kwds.get("archive_file"):
+        ctx.vlog(f"Preparing galaxy run export, history {history_id}.")
+        archive_file = kwds.get("archive_file")
+
+        jeha_id = user_gi.histories.export_history(
+            history_id=history_id,
+            wait=True,
+            maxwait=3600,
+        )
+
+        with tempfile.TemporaryDirectory() as temp_dir:
+            archive_file_output = os.path.join(temp_dir, "archive.tar.gz")
+
+            with open(archive_file_output, 'bw') as archive:
+                user_gi.histories.download_history(
+                    history_id=history_id,
+                    jeha_id=jeha_id,
+                    outf=archive
+                )
+
+            if kwds.get("arhicve_file"):
```

Comment:
There's a typo here. Can you add tests ?

Reply:
Yes, that's a type.

I have added a test for the download option.

I'm considering of writing a test that involves uploading a history to a separate Galaxy instance, and I'd appreciate some guidance on how to approach this. Would it make sense to look at the existing serve tests and try spinning up a Galaxy instance manually? Or would it be better to use the instance that Planemo starts during testing?

Reply:
Looks like I can import an invocation exported as a RO-Crate, but not when I have exported it as "Compressed File: Export the invocation to a compressed File containing the invocation data in Galaxy native format."

Reply:
@Smeds If you'd like to implement invocation export in BioBlend, it should be pretty similar to https://github.com/galaxyproject/bioblend/blob/9bdc2ce57c4443d29699ece7437d2bdf3692d4dc/bioblend/galaxy/invocations/__init__.py#L441 , I think.
Pull requests welcome!

Reply:
Interesting, the tar.gz export/import worked for me. Do you still have the archive ?

Reply:
Could be an issue with test.galaxyproject.org that I was using, seem to work on usegalaxy.org.

Is there any preference what kind of export format to use, RO-crate or Compressed Galaxy native format?

Reply:
Go with RO-crate imo, it's essentially the native format with a bit of extra metadata